geode/src/main/java/org/apache/calcite/adapter/geode/rel/GeodeSchemaFactory.java

```diff
@@ -39,18 +45,58 @@
   public static final String PDX_SERIALIZABLE_PACKAGE_PATH = "pdxSerializablePackagePath";
   public static final String ALLOW_SPATIAL_FUNCTIONS = "spatialFunction";
   public static final String COMMA_DELIMITER = ",";
+  public static final String GEODE_JNDI_INITIAL_CONTEXT_FACTORY =
+      "org.apache.geode.internal.jndi.InitialContextFactoryImpl";
```

Comment:
Why do we depend on geode internal API ? `geode.internal.jndi.InitialContextFactoryImpl`

Reply:
We would want to use ModelHandler call or just set database connection property **model**. This way the calcite connection details can still be passed as file or json string (with **inline:** prefixed) .
Will remove hard dependency of GEODE JNDI INTERNAL CONTEXT class